File: 34-find-first-and-last-position-of-element-in-sorted-array/find-first-and-last-position-of-element-in-sorted-array.py
class Solution:
    def searchRange(self, nums: List[int], x: int) -> List[int]:
        l=0
        r=len(nums)-1
        a=-1
        b=-1
        while(l<=r):
            m=(l+r)//2
            if nums[m]>=x:
                if(nums[m]==x) :
                    a=1
                r=m-1
            else :
                l=m+1
        i=r+1
        l=0
        r=len(nums)-1
        while(l<=r):
            m=(l+r)//2
            if nums[m]<=x:
                if(nums[m]==x) :
                    b=1
                l=m+1
            else:
                 r=m-1
        j=l-1
        return [-1 if a == -1 else i, -1 if b == -1 else j]

File: 34-find-first-and-last-position-of-element-in-sorted-array/find-first-and-last-position-of-element-in-sorted-array.cpp
class Solution {
public:
    vector<int> searchRange(vector<int>& nums, int target) {
        map<int,int>m;
        int smaller=0;
        for(auto x:nums){
            if(x<target) smaller+=1;
            if(x==target) m[x]+=1;
        }
        vector<int>res;
        if(m.find(target)==m.end()) return {-1,-1};
        res.push_back(smaller);
        res.push_back(smaller+m[target]-1);   
        return res;
    }
};